static/js/recent_topics.js

```diff
@@ -117,18 +117,40 @@ function set_table_focus(row, col) {
         return true;
     }
 
-    // Setting focus after the render is complete doesn't partially hide the row from view.
-    setTimeout(() => {
-        topic_rows.eq(row).find(".recent_topics_focusable").eq(col).children().trigger("focus");
-    }, 0);
+    const topic_row = topic_rows.eq(row);
+    topic_row.find(".recent_topics_focusable").eq(col).children().trigger("focus");
+
+    // Bring the focused element in view in the smoothest
+    // possible way. Using `block: center` is not a
+    // smooth scrolling experince.
+    // Using {block: "nearest"}, the element:
+    // * is aligned at the top of its ancestor if you're currently below it.
+    // * is aligned at the bottom of its ancestor if you're currently above it.
+    // * stays put, if it's already in view
+    // NOTE: Although, according to
+    // https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView#browser_compatibility
+    // `scrollIntoView` is not fully supported on Safari,
+    // it works as intended on Safari v14.0.3 on macOS Big Sur.
+    topic_row.get()[0].scrollIntoView({
+        block: "nearest",
+    });
+
     current_focus_elem = "table";
 
-    const topic_row = topic_rows.eq(row);
     const message = {
         stream: topic_row.find(".recent_topic_stream a").text(),
         topic: topic_row.find(".recent_topic_name a").text(),
     };
     compose_closed_ui.update_reply_recipient_label(message);
+
+    // focused topic can be under table `thead`
+    // or under compose, so, to avoid that
+    // from happening, we bring the element to center.
+    if (!is_topic_visible_to_user(topic_row)) {
+        topic_row.get()[0].scrollIntoView({
+            block: "center",
+        });
+    }
```

Comment:
I don't understand this block -- doesn't it call the `scrollIntoView` the same way we do earlier in this function, with no major state changes in between?  

Maybe the bug is that we should delete the first copy, since we only want to scroll if it's no visible?

Reply:
Actually the difference between `nearest` and `center` visually is that the topic row can be hidden under `thead` or `compose` when we scroll first using `nearest` and if that happens, we try to scroll it into the `center` of the scroll view to make sure it is not hidden by either top or bottom elements.

Reply:
Oh, I hadn't noticed that we had `nearest` vs `center` for the two.  OK.